sdk/servicebus/azure-servicebus/tests/stress_tests/stress_test_base.py

```diff
@@ -110,6 +133,18 @@ def PreProcessMessageBody(self, payload):
         return payload
 
 
+    def _ScheduleIntervalLogger(self, end_time, description="", interval_seconds=30):
+        def _doIntervalLogging():
+            if end_time > datetime.utcnow():
+                self._state.PopulateProcessStats()
+                _logger.critical("{} RECURRENT STATUS:".format(description))
+                _logger.critical(self._state)
+                self._ScheduleIntervalLogger(end_time, description, interval_seconds)
+
+        t = threading.Timer(interval_seconds, _doIntervalLogging)
+        t.start()
```

Comment:
would the thread t keep running if we have `self.fail_on_exception = True` and the main send/receiver thread stopped due to error?

Reply:
Solid thinking++.  I'll put a cancellation token in a finally block for the primary operation and incorporate that into the watchdog circuit breaker.